clientd/src/lib.rs

```diff
@@ -12,6 +12,32 @@ use tokio::sync::{broadcast, mpsc, oneshot};
 mod rpc;
 pub mod server;
 
+#[derive(Clone)]
+pub struct EventSubscribers(HashMap<EventKey, broadcast::Sender<Event>>);
+
+impl EventSubscribers {
+    pub fn new() -> Self {
+        let mut se_map = HashMap::new();
+        for event in EventKey::iter() {
+            let (tx, _rx) = broadcast::channel(128);
+            se_map.insert(event, tx);
+        }
+        Self(se_map)
+    }
+}
+
+impl Default for EventSubscribers {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl AsRef<HashMap<EventKey, broadcast::Sender<Event>>> for EventSubscribers {
+    fn as_ref(&self) -> &HashMap<EventKey, broadcast::Sender<Event>> {
+        &self.0
+    }
+}
```

Comment:
Not sure if that's the optimal approach, but certainly works. Passing through methods can be annoying but is afaik the rustic way? @dpc

Reply:
This type is kind of weird. Either it is supposed to encapsulate the map, in which case it should have methods for stuff that can be done with it (like sending message to subscribers for a given key), or it just a bare newtype in which case the inside could be `pub` and anyone can `.0.get(...)` etc.

Reply:
My intention was to hide the Map so I can be sure that it was created correctly and not altered/moved after that. I need at least a ref to use it though. So you would suggest dont impl `AsRef` but just add a public `send` fn ?

Reply:
you answered that below so all clear: https://github.com/fedimint/clientd/pull/2#discussion_r1036545070